ui/lib/apps/SlowQuery/utils/tableColumns.tsx

```diff
@@ -125,150 +35,115 @@ function successColumn(
-  const capacity = rows ? max(rows.map((v) => v.memory_max)) ?? 0 : 0
-  const key = 'Mem_max'
-  return {
-    name: commonColumnName('memory_max'),
-    key,
-    fieldName: 'memory_max',
-    minWidth: 140,
-    maxWidth: 200,
-    columnActionsMode: ColumnActionsMode.clickable,
-    onRender: (rec) => (
-      <Bar textWidth={70} value={rec.memory_max} capacity={capacity}>
-        {getValueFormat('bytes')(rec.memory_max, 1)}
-      </Bar>
-    ),
-  }
-}
-
-function txnStartTsColumn(
-  _rows?: { txn_start_ts?: number }[] // used for type check only
-): IColumn {
-  return {
-    name: commonColumnName('txn_start_ts'),
-    key: 'Txn_start_ts',
-    fieldName: 'txn_start_ts',
-    minWidth: 100,
-    maxWidth: 150,
-    onRender: (rec) => (
-      <Tooltip title={rec.txn_start_ts}>
-        <TextWrap>{rec.txn_start_ts}</TextWrap>
-      </Tooltip>
-    ),
+    name: tcf.columnName(fieldName),
+    key: fieldName,
+    fieldName: fieldName,
+    minWidth: 50,
+    maxWidth: 100,
+    onRender: (rec) => (rec.is_internal === 1 ? 'Yes' : 'No'),
   }
 }
 
 //////////////////////////////////////////
 
 export function slowQueryColumns(
-  rows: SlowqueryBase[],
+  rows: SlowquerySlowQuery[],
   showFullSQL?: boolean
 ): IColumn[] {
+  const tcf = new TableColumnFactory(TRANS_KEY_PREFIX)
   return [
-    sqlColumn(rows, showFullSQL),
-    digestColumn(rows),
-    instanceColumn(rows),
-    dbColumn(rows),
-    timestampColumn(rows),
-    queryTimeColumn(rows),
-    parseTimeColumn(rows),
-    compileTimeColumn(rows),
-    processTimeColumn(rows),
-    memoryColumn(rows),
-    txnStartTsColumn(rows),
-    successColumn(rows),
+    tcf.sqlText('query', showFullSQL),
+    tcf.textWithTooltip('digest'),
+    tcf.textWithTooltip('instance'),
+    tcf.textWithTooltip('db'),
+    tcf.textWithTooltip('connection_id'),
+    tcf.timestamp('timestamp'),
+
+    tcf.bar.single('query_time', 's', rows),
+    tcf.bar.single('parse_time', 's', rows),
+    tcf.bar.single('compile_time', 's', rows),
+    tcf.bar.single('process_time', 's', rows),
+    tcf.bar.single('memory_max', 'bytes', rows),
+
+    tcf.textWithTooltip('txn_start_ts'),
+    successColumn(tcf, rows),
+    // basic
+    isInternalColumn(tcf, rows),
+    tcf.textWithTooltip('index_names'),
+    tcf.textWithTooltip('stats'),
+    tcf.textWithTooltip('backoff_types'),
+    // connection
+    tcf.textWithTooltip('user'),
+    tcf.textWithTooltip('host'),
+    // time
+    tcf.bar.single('wait_time', 'ns', rows),
+    tcf.bar.single('backoff_time', 'ns', rows),
+    tcf.bar.single('get_commit_ts_time', 'ns', rows),
+    tcf.bar.single('local_latch_wait_time', 'ns', rows),
+    tcf.bar.single('prewrite_time', 'ns', rows),
+    tcf.bar.single('commit_time', 'ns', rows),
+    tcf.bar.single('commit_backoff_time', 'ns', rows),
+    tcf.bar.single('resolve_lock_time', 'ns', rows),
+    // cop
+    tcf.bar.multiple(
+      {
+        avg: { fieldName: 'cop_proc_avg', tooltipPrefix: 'Mean:' },
```

Comment:
This can be further simplified by moving `tooltipPrefix` to the key, like: `{ mean: 'cop_proc_avg' }`